### ISSUE TITLE
Ormlite - Case Sensitivity Issue w/ OrmLiteNamingStrategyBase

### ISSUE THREAD

**Issue by eweast** [ServiceStack/Issues#390]
I'm using Postgres and made the unfortunate decision to utilize capitalization in both table and column names. However, aside from this one issue, I haven't had any other case sensitivity issues using `OrmLiteNamingStrategyBase()` with selects, inserts, deletes, or updates.

I'm getting "column myid does not exist" exceptions when attempting to use either of the below methods to obtain the last auto-incremented Id on insert. The issue is that the column's name is "MyId" so it appears case isn't being preserved which is causing the exception.

```c#
class Poco
{
    public string Text { get; set; }
    public int MyId { get; set; } 
}

//#1
var rowId = db.Insert(new Poco{ Text = "Text" }, selectIdentity:true);

//#2
var obj = new Poco() { Text = "Text"};
Db.Save(obj);
```

**Comment by mythz** [comment]
Is this suppose to be a failing test case? did you create the table with `db.CreateTable<Poco>()`?

**Comment by eweast** [comment]
No, the table was created by other tools and I used OrmLite.T4 to generate the classes for the tables.

I over simplified my ealier example, the actual Poco class would be... 

```c#
public partial class Poco
 {
    [AutoIncrement]
    [PrimaryKey]
    public int MyId { get; set;}
    [Required]
    public string Text { get; set;}
}
```

The exception....
```
Npgsql.NpgsqlException was unhandled by user code
  HResult=-2147467259
  Message=42703: column "myid" does not exist
  Source=Npgsql
  ErrorCode=-2147467259
  BaseMessage=column "myid" does not exist
  Code=42703
  File=parse_relation.c
  InternalPosition=0
  Line=2892
  MessageText=column "myid" does not exist
  Position=190
  Routine=errorMissingColumn
  Severity=ERROR
  StackTrace:
       at Npgsql.NpgsqlConnector.DoReadSingleMessage(DataRowLoadingMode dataRowLoadingMode, Boolean returnNullForAsyncMessage, Boolean isPrependedMessage)
       at Npgsql.NpgsqlConnector.ReadSingleMessage(DataRowLoadingMode dataRowLoadingMode, Boolean returnNullForAsyncMessage)
       at Npgsql.NpgsqlCommand.Execute(CommandBehavior behavior)
       at Npgsql.NpgsqlCommand.ExecuteScalarInternal()
       at ServiceStack.OrmLite.OrmLiteReadCommandExtensions.LongScalar(IDbCommand dbCmd)
       at ServiceStack.OrmLite.OrmLiteWriteCommandExtensions.Save[T](IDbCommand dbCmd, T obj)
       at ServiceStack.OrmLite.OrmLiteExecFilter.Exec[T](IDbConnection dbConn, Func`2 filter)
...
```

**Comment by mythz** [comment]
So then the issue is an invalid mapping, i.e. your DB column names are not the same as what OrmLite expects based on its default naming convention.

You can change OrmLite's default naming conventions with:

```csharp
PostgreSqlDialect.Provider.NamingStrategy = new OrmLiteNamingStrategyBase();
```

Here are some other [built-in NamingStrategies](https://github.com/ServiceStack/ServiceStack.OrmLite/blob/master/src/ServiceStack.OrmLite/NamingStrategy.cs) whilst the [Naming Strategy Tests](https://github.com/ServiceStack/ServiceStack.OrmLite/blob/master/tests/ServiceStack.OrmLite.Tests/OrmLiteCreateTableWithNamingStrategyTest.cs) show more examples of implementing custom Naming Strategies by implementing `INamingStrategy`.

Otherwise if you want to override the default naming convention you need to use `[Alias]`

**Comment by eweast** [comment]
I identified the issue and it's the SQL emitted when```selectIdentity``` is```true```.

Using ```OrmLiteNamingStrategyBase``` naming strategy, ```
Db.Insert(new Poco{ Text = "Text" });``` works without error and emits ```
INSERT INTO "Poco" ("MyId","Text") VALUES (:MyId,:Text)``` The ```OrmLiteNamingStrategyBase``` works here and for Select, Update, and Delete methods as well.

However,
```c#
Db.Insert(new Poco{ Text = "Text" }, selectIdentity:true);
```
emits...
```sql
INSERT INTO "Poco" ("MyId","Text") VALUES (:MyId,:Text) RETURNING MyId
```

```MyId``` after ```RETURNING``` is capitalized correctly, but is missing the quotation marks so Postgres will return it as ```myid```, thus the mapping error. The naming strategies and Alias attribute don't seem to change how the identity after ```RETURNING``` is emitted.

**Comment by mythz** [comment]
Should be resolved with this commit: https://github.com/ServiceStack/ServiceStack.OrmLite/commit/43ec5e915f3bddf8eb3acc74c2857456d8c5ac85

This change is available from v4.0.53 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

**Comment by eweast** [comment]
That did the trick.

Thanks Demis!
